test/types/Model.ts

```diff
@@ -9,3 +10,30 @@ const shouldFailWithNumberAsName = dynamoose.model(1);
 const shouldFailWithBooleanAsName = dynamoose.model(true);
 
 const shouldSucceedWithOnlyPassingInName = dynamoose.model("User");
+
+interface User extends Document {
+	id: string;
+	name: string;
+}
+
+const UserModel = dynamoose.model<User>(
+	"User",
+	new dynamoose.Schema({
+		"id": String,
+		"name": {
+			"type": String,
+			"index": {
+				"global": true
+			}
+		},
+		"age": Number
+	})
+);
+
+UserModel.create({"id": "1", "name": "john", "age": 25});
```

Comment:
I think that all of these should have test names for them. For example.

```suggestion
const shouldCreateSuccessfully = TypedModel.create({"id": "1", "name": "john", "age": 25});
```

Reply:
I think we should also have a failing test here. Shouldn't the following fail?

```js
const shouldNotFailWithNotAllAttributesPassedIn = TypedModel.create({"id": "1"});
// @ts-expect-error
const shouldFailWithInvalidAttributes = TypedModel.create({"id": "1", "random": "string"});
```

Reply:
currently create() still don't take the generic type, my plan is in next phase, since there are too much changes.

Reply:
@hardyscc Does anything take the generic type yet? Or is this PR just preparation for that?